### ISSUE TITLE
bug: ion-item-divider end slot has no padding

### ISSUE THREAD

**Issue by infacto** [ionic-team/ionic-framework#23785]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

The `end` slot is too close to the frame. It has no padding as the default slot.

![image](https://user-images.githubusercontent.com/60390085/129884903-2bf44c99-4879-4d94-9dff-51a9c1581cd7.png)

### Expected Behavior

The `end` slot should also have the same padding as the default slot. In other words: The `ion-item-divider` should have symmetric horizontal padding.

![image](https://user-images.githubusercontent.com/60390085/129884821-90e96ca0-2d3a-4968-93bc-754c71da6f55.png)

### Steps to Reproduce

```html
<ion-list class="ion-no-padding">
  <ion-item-divider>
    Items
    <span slot="end">0/4</span>
  </ion-item-divider>
</ion-list>
```

[See CodePen](https://codepen.io/infacto/pen/poPXRGx)

### Ionic Info

(It's a native Angular project without Ionic CLI. So info does not work.)
"@ionic/angular": "^5.6.7"

### Additional Information

![demo](https://user-images.githubusercontent.com/60390085/129884567-84607bc3-b7cd-4ebd-8a6a-5232409671dd.gif)

`<ion-item-divider>`: 
![image](https://user-images.githubusercontent.com/60390085/129885003-8e244eb7-98ae-4e55-b8bb-76e2251e10cf.png)

So just add the same padding on the right. Voila.

Workaround (global styles): 

```css
ion-item-divider {
  padding: 0 16px;
}
```
But this should handle the framework. Or is there a reason why there is no padding?

Another related issue: `div.item-divider-inner` should not use `overflow: hidden`. It affects the end-slot to be cropted if moved (e.g. with transform). Because the padding comes from the parent, this element has the width minus padding and cops the text moved a bit outside. We could overflow the host?

The structure of the elements is a bit weird. Slot "start" is outside of ".item-divider-inner" And end slot is part of inner. etc.

**Comment by liamdebeasi** [comment]
Thanks, I can reproduce this issue. Looks like the start and end padding values for the divider in iOS mode uses the same padding as `ion-item`, but the divider in MD mode only uses the padding for `ion-item` for the start padding.

**Comment by infacto** [comment]
I also wonder why this special layout? [See source](https://github.com/ionic-team/ionic-framework/blob/v5.6.14/core/src/components/item-divider/item-divider.tsx#L52-L58)

```html
<slot name="start"></slot>
<div class="item-divider-inner">
  <div class="item-divider-wrapper">
    <slot></slot>
  </div>
  <slot name="end"></slot>
</div>
```

- Slot "start" is outside the inner div.
- Slot "end" and main slot is inside the inner div.
- The main slot is wrapped again in a wrapper.
- etc.

It's not the most important component. But maybe requires a code redesign?

**Comment by mapsandapps** [comment]
Thanks you for submitting this issue. It has been fixed in #27019, and a fix will be available in an upcoming release of Ionic Framework.
